### ISSUE TITLE
RANSAC Randomization 

### ISSUE THREAD

**Issue by YoussephAhmed** [colmap/colmap#757]
I knew that on the commit b738bc8e96 the randomization and output consistency problem should be solved.. but I think I still got some problem with random output, so is there a seed option or parameter where I can fix it or something? and thank you so much for your quick response and feedback.

**Comment by YoussephAhmed** [comment]
Also is the sequence of cli commands here is right for a sparse reconstruction? :
1) colmap feature_extractor
2) colmap sequential_matcher 
3) colmap mapper 

and the problem in the randomization that in mapper stage it's terminating to fast with no convergence.. without a construction and maybe in a next run (same data & same parameters for the command options) it may result a sparse reconstruction model so any hints about the parameter that can fix this random issue ?

**Comment by ahojnnes** [comment]
The pipeline is only deterministic for the mapper stage. The matching is still non-deterministic. So, if you run mapper repeatedly on the same database, it should produce consistent results, but not if you re-run any of the *_matchers.

**Comment by YoussephAhmed** [comment]
Yes I can see this.. is there anyway to reduce this random behavior like maybe setting --SiftMatching.guided_matching 0 or something or are you working on making it deterministic soon? and again thank you so much.

**Comment by ahojnnes** [comment]
Closing as this is by design — the mapper is deterministic but matchers are intentionally non-deterministic, as explained by the maintainer.
